tensorflow/lite/micro/kernels/conv.h

```diff
@@ -87,6 +87,16 @@ inline TfLiteRegistration Register_CONV_2D_INT8REF() {
 }
 #endif
 
+#if defined(CMSIS_NN)
+// Returns a TfLiteRegistration struct for kernel variant that only supports
+// int16 inputs and outputs.
+TfLiteRegistration Register_CONV_2D_INT16X8REF();
```

Comment:
That the INT8 variant has REF in its name is because it is a way to explicitly call the reference implementation even when we select OPTIMIZED_KERNEL_DIR=xtensa.

In the case of this PR, what you are looking for are variants called CONV_2D_INT8 and CONV_2D_INT16.

Similar to what we have for fully connected:
https://github.com/tensorflow/tflite-micro/blob/1c6b50ef8b9dac1bb31dfe79679aec781666a86b/tensorflow/lite/micro/kernels/fully_connected.h#L68-L81

I would suggest the following

```cc
#if defined(XTENSA)
// Returns a TfLiteRegistration struct for kernel variant that only supports
// int8 activations and int8 weights and always calls the reference implementation.
TfLiteRegistration Register_CONV_2D_INT8REF();
#else
inline TfLiteRegistration Register_CONV_2D_INT8REF() {
  return Register_CONV_2D();
}

#if defined(CMSIS_NN)
// Returns a TfLiteRegistration struct for kernel variant that only supports
// int8 activations and int8 weights and uses the latency optimized implementations.
TfLiteRegistration Register_CONV_2D_INT8();

// Returns a TfLiteRegistration struct for kernel variant that only supports
// int16 activations and int8 weights and uses the latency optimized implementations.
TfLiteRegistration Register_CONV_2D_INT16();

#else
inline TfLiteRegistration Register_CONV_2D_INT8() {
  return Register_CONV_2D();
}

inline TfLiteRegistration Register_CONV_2D_INT16() {
  return Register_CONV_2D();
}

```

I'm OK with changing person_detect to use the CONV_2D_INT8 for all targets.

Reply:
Ok got it.